xdebug.c

```diff
@@ -1937,9 +1946,15 @@ void xdebug_execute_ex(zend_execute_data *execute_data TSRMLS_DC)
 	}
 
 	/* Check for entry breakpoints */
-	if (xdebug_is_debug_connection_active_for_current_pid() && XG(breakpoints_allowed)) {
-		if (!handle_breakpoints(fse, XDEBUG_BREAKPOINT_TYPE_CALL)) {
-			xdebug_mark_debug_connection_not_active();
+	if(XG(breakpoints_allowed)) {
+		block = 0;
+		if (xdebug_is_debug_connection_active_for_current_pid()) {
+			if (!handle_breakpoints(fse, XDEBUG_BREAKPOINT_TYPE_CALL, &block)) {
+				xdebug_mark_debug_connection_not_active();
+			}
+		}
+		if (xdebug_is_debug_connection_active_for_current_pid()) {
```

Comment:
What is the reason for doing this twice?

Reply:
I'm not sure if `handle_breakpoints` failed (which cause `xdebug_mark_debug_connection_not_active`). So, check `xdebug_is_debug_connection_active_for_current_pid` again after `handle_breakpoints`.